File: examples/benchmarks/oo5.py
class Foo:
    def __init__(self):
	self.i = 10
	self.j = 20

    def I(self):
	return self.i
    def J(self):
	return self.j

class main:
    o = Foo()
    for i in range(1,500001):
	x = o.I()
	y = o.J()
    print o.I()

File: examples/benchmarks/oo5.pl
#! perl -w
use strict;

my $o = new Foo();
for my $i (1 .. 500000) {
	my $x = $o->i;
	my $y = $o->j;
}
print $o->i, "\n";

package Foo;

sub new {
    my $self = ref $_[0] ? ref shift : shift;
    return bless [ 10, 20 ], $self;
}

sub i {
    my $self = shift;
    if (@_) { $self->[0] = shift } else { $self->[0] }
}

sub j {
    my $self = shift;
    if (@_) { $self->[1] = shift } else { $self->[1] }
}

1;
